boards/arm/mimxrt595_evk/mimxrt595_evk_cm33.dts

```diff
@@ -368,23 +362,19 @@ zephyr_udc0: &usbhs {
 
 			boot_partition: partition@0 {
 				label = "mcuboot";
-				reg = <0x00000000 DT_SIZE_K(64)>;
+				reg = <0x00000000 DT_SIZE_K(256)>;
 			};
-			slot0_partition: partition@10000 {
+			slot0_partition: partition@40000 {
 				label = "image-0";
-				reg = <0x00010000 DT_SIZE_M(24)>;
+				reg = <0x00040000 DT_SIZE_K(3076)>;
```

Comment:
what if the image is larger than 3 MB?

Reply:
Then the user will need to change the partition layout. These partitions are only a suggestion, and I have deliberately kept them small to minimize upgrade time with MCUBoot. MCUBoot does not seem to have awareness of the true size of an image, only the size of the slot, so if using 24MB slots, MCUBoot will attempt to copy 48MB of data during a swap upgrade procedure (most of it zeros). This results in a very slow upgrade time, which impacts the ability to test support for this feature

Reply:
That sounds like a bug in your generation script. Not that I know off the top of my head if mcuboot copies empty data or not, but mcuboot will absolutely copy zeros, the only data it might skip would be flash erased values, i.e. 0xff